azurelinuxagent/common/protocol/wire.py

```diff
@@ -873,34 +926,61 @@ def get_certs(self):
             return None
         return self.certs
 
-    def get_current_handlers(self):
-        handler_list = list()
-        try:
-            incarnation = self.fetch_cache(os.path.join(conf.get_lib_dir(),
-                                                        INCARNATION_FILE_NAME))
-            ext_conf = ExtensionsConfig(self.fetch_cache(os.path.join(conf.get_lib_dir(),
-                                                                      EXT_CONF_FILE_NAME.format(incarnation))))
-            handler_list = ext_conf.ext_handlers.extHandlers
-        except ProtocolError as pe:
-            # cache file is missing, nothing to do
-            logger.verbose(ustr(pe))
-        except Exception as e:
-            logger.error("Could not obtain current handlers: {0}", ustr(e))
-
-        return handler_list
+    def get_sequence_number(self):
+        path = os.path.join(conf.get_lib_dir(), SEQUENCE_NUMBER_FILE_NAME)
+        if os.path.exists(path):
+            return int(fileutil.read_file(path))
```

Comment:
`fileutil.read_file` can also return a None in certain scenarios (if there is some error in reading the file). Casting `int(None)` would throw an exception. you should handle that case